### PR TITLE
[QPG] Adding ota-requestor support in qpg lighting app

### PR DESCRIPTION
**Pull request by tima-q**
#### Problem
- No OTA platform support in QPG platform layer
- No OTA Requestor support in lighting-app

#### Change overview

QPG:
* Add OTA upgrade to QPG platform and lighting-app
* Enable DNS client to resolve OTA provider
* Cleanup button handling in light app, enabling Trigger update button.

General:
* Add ota-upgrade AnnounceOTAProvider handling to lighting-app .zap

Note:
Duplicating ota-requestor app for our platform had little merit.
The requestor app holds virtually the same clusters as the lighting-app.
Enabling OTA is now a matter of adding the underlying OTA implementation + top-level initialization as described in the OTA documentation.

Further work remaining for testing and processing received image.
Current ToT showing instabilities in terms of transfer of full image.

#### Testing
* Setup:
 RPi4 running ota-provider + chip-tool
 QPG QPG6105 DK Board running lighting app
* Test
 Start ota-provider with dummy .bin file
`chip-ota-provider-app -f /home/ubuntu/connectedhomeip/ota.bin`
 Join with chip-tool
`./chip-tool pairing onnetwork 1 20202021`
 Join QPG board
`./chip-tool pairing ble-thread 2 hex:000300000f02081111111122222222051000112233445566778899aabbccddeeff01021234 20202021 3840`
 Announce OTA image is available with chip-tool
`./chip-tool otasoftwareupdaterequestor announce-ota-provider 1 0 0 0 2 0`
* Outcome
 QPG board queries ota-provider app
 Transfer starts - triggering QPG OTA functionality

### CODE REVIEW

File: examples/platform/qpg/ota/ota.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    // Connect the Downloader and Image Processor objects
+    gDownloader.SetImageProcessorDelegate(&gImageProcessor);
+
+    gRequestorCore.SetBDXDownloader(&gDownloader);
+}
+
+void TriggerOTAQuery(void)
+{
+    static_cast<OTARequestor *>(GetRequestorInstance())->TriggerImmediateQuery();
```
Comment (Damian-Nordic) [inline examples/platform/qpg/ota/ota.cpp:81]: You don't need the cast. The method is part of the interface already.

**Review by Damian-Nordic — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: examples/platform/qpg/ota/ota.cpp (hunk 1 above)
Comment (tima-q) [reply examples/platform/qpg/ota/ota.cpp]: Addressed - thanks.

File: src/platform/qpg/OTAImageProcessorImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    {
+        imageProcessor->mDownloader->EndDownload(CHIP_ERROR_WRITE_FAILED);
+        return;
+    }
+
+    imageProcessor->mParams.downloadedBytes += imageProcessor->mBlock.size();
+    imageProcessor->mDownloader->FetchNextData();
+}
+
+CHIP_ERROR OTAImageProcessorImpl::SetBlock(ByteSpan & block)
```
Comment (selissia) [inline src/platform/qpg/OTAImageProcessorImpl.cpp:171]: Please note that there is a bug here (comes from the original Linux implementation of this this function). See https://github.com/project-chip/connectedhomeip/issues/13393 . 
There's a plan to fix that issue in all the platforms but it's currently blocked by another Linux OTA issue

**Review by selissia — APPROVED** [review]

File: examples/lighting-app/esp32/main/CMakeLists.txt [hunk 3]
```diff
@@ -48,6 +48,7 @@ idf_component_register(PRIV_INCLUDE_DIRS
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/occupancy-sensor-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/on-off-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/operational-credentials-server"
+                      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ota-requestor"
```
Comment (dhrishi) [inline examples/lighting-app/esp32/main/CMakeLists.txt:51]: Why is this change added here?

Comment (dhrishi) [reply examples/lighting-app/esp32/main/CMakeLists.txt:51]: Anyway, that change should be a part of #13532

Comment (tima-q) [reply examples/lighting-app/esp32/main/CMakeLists.txt:51]: CI does not pass if it's not updated for all platforms - This PR was prepared already before #13532 was published.

Comment (tima-q) [reply examples/lighting-app/esp32/main/CMakeLists.txt:51]: @dhrishi pinged you on Slack to discuss further if required

Comment (dhrishi) [reply examples/lighting-app/esp32/main/CMakeLists.txt:51]: >CI does not pass if it's not updated for all platforms - This PR was prepared already before #13532 was published.

@tima-q Didn't realize that CI won't pass. Thanks!
